plugins/org.eclipse.n4js/src/org/eclipse/n4js/external/NpmPackageToProjectAdapter.java

```diff
@@ -90,8 +91,26 @@ public boolean accept(File pathname) {
 		}
 	};
 
-	/** Sub folder within the N4JSD project that contains the actual N4JSD files. */
-	private final static String N4JSD_SRC_FOLDER = "src";
+	/** Default filter for copying N4JSD project contents during adaptation */
+	private final static Predicate<Path> COPY_N4JSD_PREDICATE = new Predicate<Path>() {
+		private final Path[] EXCLUDE = { Paths.get(".project"), Paths.get("manifest.n4mf"),
+				Paths.get("manifest.fragment") };
+
+		@Override
+		public boolean apply(Path path) {
+			Path filename = path.getFileName();
+			if (filename == null)
+				return false;
+
+			for (Path exclude : EXCLUDE) {
+				if (filename.equals(exclude)) {
```

Comment:
should we have filter, to allow only `n4jsd` files only? Otherwise someone can inject code in `.js` file, or sneak in other content with side effects (scripts, symlinks, binaries ?)

Reply:
Already changed.